cmake/opendds_build_helpers.cmake

```
@@ -18,6 +31,13 @@ function(_opendds_alias target)
     add_executable("${name}" ALIAS "${target}")
   endif()
   set_target_properties(${target} PROPERTIES EXPORT_NAME "${name}")
+
+  if(OPENDDS_COMPILE_WARNINGS STREQUAL "WARNING" OR OPENDDS_COMPILE_WARNINGS STREQUAL "ERROR")
+    target_compile_options(${target} PRIVATE ${CMAKE_CXX_COMPILER_ID}-warning)
```

@iguessthislldo What is the necessary syntax here?

---

I think what you want is:

```suggestion
    target_compile_options(${target} PRIVATE ${${CMAKE_CXX_COMPILER_ID}-warning)}
```

---

I'm assuming a space delimited list will be interpreted the same as a normal CMake `;`-delimited list though.